src/components/shared/chart/Viewport.js

```diff
@@ -463,6 +511,94 @@ export const withChartViewport: WithChartViewport<*, *> =
         this.moveViewport(offsetX, offsetY);
       };
 
+      _keyDownListener = (
+        event: { nativeEvent: KeyboardEvent } & SyntheticKeyboardEvent<>
+      ) => {
+        let navigationKey;
+        if (
+          !event.ctrlKey &&
+          !event.shiftKey &&
+          !event.altKey &&
+          !event.metaKey
+        ) {
+          navigationKey = BARE_KEYMAP[event.nativeEvent.code];
+        } else if (event.ctrlKey) {
+          navigationKey = CTRL_KEYMAP[event.nativeEvent.code];
+        }
+
+        if (navigationKey !== undefined) {
+          // Start requesting frames if there were no keys down before
+          // this event triggered.
+          if (this._keysDown.size === 0) {
+            requestAnimationFrame(this._keyboardNavigation);
+            this._lastKeyboardNavigationFrame = performance.now();
+          }
+
+          this._keysDown.add(navigationKey);
+          event.preventDefault();
+        }
+      };
+
+      _keyUpListener = (
+        event: { nativeEvent: KeyboardEvent } & SyntheticKeyboardEvent<>
+      ) => {
+        if (!event.ctrlKey) {
+          // The ctrl modifier might have been released here. Try to
+          // delete all keys associated with the modifier. Since the
+          // navigation is aliased with non-ctrl-modified keys also,
+          // this will affect (stop) the operation even if it was
+          // introduced without a ctrl-modified key.
+          for (const code of Object.keys(CTRL_KEYMAP)) {
+            this._keysDown.delete(CTRL_KEYMAP[code]);
+          }
+        }
+        this._keysDown.delete(CTRL_KEYMAP[event.nativeEvent.code]);
+        this._keysDown.delete(BARE_KEYMAP[event.nativeEvent.code]);
+      };
+
+      _keyboardNavigation = timestamp => {
+        if (this._keysDown.size === 0) {
+          // No keys are down, nothing to do.  Don't request a new
+          // animation frame.
+          this._lastKeyboardNavigationFrame = null;
+          return;
+        }
+        requestAnimationFrame(this._keyboardNavigation);
+
+        const dt =
+          this._lastKeyboardNavigationFrame === null
+            ? 0
+            : timestamp - this._lastKeyboardNavigationFrame;
+        const delta = KEYBOARD_NAVIGATION_SPEED * dt * 0.001;
+        this._lastKeyboardNavigationFrame = timestamp;
+
+        for (const k of this._keysDown.values()) {
+          const navigationKey: NavigationKey = k;
```

Comment:
This is also an ugly workaround to keep Flow happy. Is there a better way to accomplish this?

Reply:
I must be turning it into a `string` and not respecting the union. You could create a function in `src/utils/flow.js` that could enforce this constraint. Otherwise you might be able to do `switch ((navigationKey: NavigationKey)) {